File: WS/DSTests/algos/binary_heap/priority_queue_test.py
from unittest import TestCase

from DS.algos.binary_heap.priority_queue import PriorityQueue, MAX_PQ, MIN_PQ


class PriorityQueueTest(TestCase):
    def setUp(self):
        self.max_pq = PriorityQueue(10, MAX_PQ)
        self.min_pq = PriorityQueue(10, MIN_PQ)

    def test_max_pq(self):
        self.max_pq.insert(6)
        self.max_pq.insert(1)
        self.max_pq.insert(2)
        self.max_pq.insert(3)

        self.assertEqual(6, self.max_pq.delete_top())
        self.assertEqual(3, self.max_pq.delete_top())
        self.assertEqual(2, self.max_pq.delete_top())
        self.assertEqual(1, self.max_pq.delete_top())

    def test_min_pq(self):
        self.min_pq.insert(6)
        self.min_pq.insert(1)
        self.min_pq.insert(2)
        self.min_pq.insert(3)

        self.assertEqual(1, self.min_pq.delete_top())
        self.assertEqual(2, self.min_pq.delete_top())
        self.assertEqual(3, self.min_pq.delete_top())
        self.assertEqual(6, self.min_pq.delete_top())

File: ws/src/test/java/workspace/ws/ds/algos/heaps/PriorityQueueTest.java
package workspace.ws.ds.algos.heaps;

import junit.framework.TestCase;

public class PriorityQueueTest extends TestCase {
	private MaxPriorityQueue<Integer> mpq;
	private MinPriorityQueue<Integer> mpq2;
	
	protected void setUp() {
		mpq = new MaxPriorityQueue<Integer>(10);
		mpq2 = new MinPriorityQueue<Integer>(10);
	}

	public void testMaxPriorityQueue() {
		mpq.insert(3);
		mpq.insert(6);
		mpq.insert(2);
		mpq.insert(1);
		
		int max = mpq.deleteTop();
		assertEquals(6, max);
		max = mpq.deleteTop();
		assertEquals(3, max);
		max = mpq.deleteTop();
		assertEquals(2, max);
		max = mpq.deleteTop();
		assertEquals(1, max);
	}
	
	public void testMinPriorityQueue() {
		mpq2.insert(3);
		mpq2.insert(6);
		mpq2.insert(2);
		mpq2.insert(1);
		
		int max = mpq2.deleteTop();
		assertEquals(1, max);
		max = mpq2.deleteTop();
		assertEquals(2, max);
		max = mpq2.deleteTop();
		assertEquals(3, max);
		max = mpq2.deleteTop();
		assertEquals(6, max);
	}
}
